test/filtering.test.js

```diff
@@ -252,6 +262,34 @@ describe('array data provider', () => {
     expect(getBodyCellContent(grid, 0, 0).innerText).to.equal('bar');
   });
 
+  // TODO: find better option to check rather than grid._cache.items...
+  it('should update filtering when column is removed', () => {
+    filterFirst.value = 'bar';
+    flushFilters(grid);
+
+    grid.removeChild(grid.firstElementChild);
+    flushGrid(grid);
+
+    expect(Object.keys(grid._cache.items).length).to.equal(3);
+  });
+
+  // TODO: find better option to check rather than grid._cache.items...
```

Comment:
In case grid is not attached, I guess we can proceed with checking `_cache` or alternatively spy on i.e. updating the item.

Reply:
Kept code and removed TODO